src/main/java/com/github/lookout/metrics/agent/generators/YammerMetricsGenerator.java

```diff
@@ -0,0 +1,87 @@
+package com.github.lookout.metrics.agent.generators;
+
+import com.timgroup.statsd.StatsDClient;
+import com.yammer.metrics.Metrics;
+import com.yammer.metrics.core.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by rkuris on 5/28/15.
+ */
+public class YammerMetricsGenerator implements MetricProcessor<Long>, MetricGenerator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(YammerMetricsGenerator.class);
+    private MetricsRegistry registry = Metrics.defaultRegistry();
+    private StatsDClient client;
+    private static final MetricPredicate predicate = MetricPredicate.ALL;
+    protected final Clock clock = Clock.defaultClock();
+
+    @Override
+    public void generate(StatsDClient client) {
+        // get all the registered metrics via dropwizard
+        this.client = client;
+        final long epoch = TimeUnit.MILLISECONDS.toSeconds(clock.time());
+        final Set<Map.Entry<String, SortedMap<MetricName, Metric>>> entries = registry.groupedMetrics(predicate).entrySet();
+
+        for (final Map.Entry<String, SortedMap<MetricName, Metric>> entry : entries) {
+            for (final Map.Entry<MetricName, Metric> subEntry : entry.getValue().entrySet()) {
+
+                final Metric metric = subEntry.getValue();
+
+                if (metric != null) {
+                    try {
+                        metric.processWith(this, subEntry.getKey(), epoch);
+                    } catch (final Exception exception) {
+                        LOG.error("Error processing key {}", subEntry.getKey(), exception);
+                    }
+                }
+            }
+        }
+
+    }
+    @Override
+    public void processMeter(MetricName name, Metered meter, Long epoch) throws Exception {
+        LOG.debug("Meter {} {} skipped", name.getName(), meter.count());
+    }
+
+    @Override
+    public void processCounter(MetricName name, Counter counter, Long epoch) throws Exception {
+        client.gauge(name.getName(), counter.count());
```

Comment:
As you know, objectName consists of domain:properties. Each property consists of the type=value pairs. 

name.getName() only returns value for the "name", a very short name, doesnt tell what MBean it belongs to.
    'thq-m-mkelk01.MinRowSize': 20446,
    'thq-m-mkelk01.MeanRowSize': 69684,

I was wondering if we should use name.getMBeanName() instead of name.getName(). But, alas, it contains special characters that statsd don't like. 

jmxtrans-agent will allow you to use full MBean name with format palatable to statsd